tests/unit-tests/CheckoutProcess.php

```diff
@@ -0,0 +1,75 @@
+<?php
+// phpcs:ignoreFile
+
+namespace GoogleAnalyticsIntegration\Tests;
+
+use MockAction;
+use WP_UnitTestCase;
+use WC_Helper_Product;
+use WC_Helper_Customer;
+use WC_Google_Gtag_JS;
+
+class CheckoutProcessTest extends WP_UnitTestCase {
+
+	public function test_begin_checkout_event() {
+		$product  = WC_Helper_Product::create_simple_product();
+		$customer = WC_Helper_Customer::create_customer( 'JD', 'pw', 'customer@unit.test' );
+
+		wp_set_current_user( $customer->get_id() );
+
+		$cart = WC()->cart;
+		$cart->add_to_cart( $product->get_id() );
+
+		// Mock woocommerce_gtag_event_data filter to ensure it is called and the correct data is processed
+		$filter = new MockAction();
+		add_filter( 'woocommerce_gtag_event_data', array( &$filter, 'filter' ) );
+
+		( new WC_Google_Gtag_JS() )->checkout_process( $cart->get_cart() );
+
+		// Confirm woocommerce_gtag_event_data is called by checkout_process()
+		$this->assertEquals( 1, $filter->get_call_count(), 'woocommerce_gtag_event_data filter was not called for begin_checkout (checkout_process()) event' );
+
+		// The expected data structure for this event
+		$expected_data = array(
+			'items' => array(
+				array(
+					'id'       => WC_Google_Gtag_JS::get_product_identifier( $product ),
```

Comment:
I agree we need to expand test coverage. For now my focus, like you said, was purely on testing the data structure for events. I've opened issue #248 so that we can look to expand coverage down the road.